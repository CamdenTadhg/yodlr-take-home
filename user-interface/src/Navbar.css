#Navbar {
    background-color: #643A71;
    height: 10vh;
    font-size: 1.25rem;
}

#Navbar-logo {
    margin: 10px;
    margin-left: 20px;
    color: white;
    float: left;
    padding-top: 10px;
}

#Navbar-logo:hover {
    color: #FEC0CE
}

.Navbar-navlink {
    margin: 10px;
    color: white;
    float: right;
    padding-top: 10px;
    margin-right: 20px;
}

.Navbar-navlink:hover {
    color: #FEC0CE
}
