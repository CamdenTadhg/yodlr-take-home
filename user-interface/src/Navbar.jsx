import React from 'react';
import {Link} from 'react-router-dom';
import './Navbar.css';

const Navbar = () => {
    return (
        <div id='Navbar'>
            <Link id='Navbar-logo' to='/'>Yodlr</Link>
            <Link className='Navbar-navlink' to='/signup'>Signup</Link>
            <Link className='Navbar-navlink' to='/admin'>Admin</Link> 
        </div>
    );
};

export default Navbar;