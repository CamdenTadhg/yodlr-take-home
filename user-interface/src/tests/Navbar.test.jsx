import React from 'react';
import {render, getByText} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom';
import {it, expect} from 'vitest';
import Navbar from '../Navbar'


it('renders without crashing', () => {
    render(
    <MemoryRouter>
        <Navbar/>
    </MemoryRouter>);
});

it('matches the snapshot', () => {
    const {asFragment} = render(
        <MemoryRouter>
            <Navbar/>
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it('displays the correct content', () => {
    render(
        <MemoryRouter>
            <Navbar/>
        </MemoryRouter>
    );

    expect(getByText(/Admin/i)).toBeInTheDocument();
    expect(getByText(/Yodlr/i)).toBeInTheDocument();
    expect(getByText(/Signup/i)).toBeInTheDocument();
})