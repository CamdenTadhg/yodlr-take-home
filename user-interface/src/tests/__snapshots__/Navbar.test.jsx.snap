// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    id="Navbar"
  >
    <a
      href="/"
      id="Navbar-logo"
    >
      Yodlr Home
    </a>
    <a
      class="Navbar-navlink"
      href="/signup"
    >
      Signup
    </a>
    <a
      class="Navbar-navlink"
      href="/admin"
    >
      Admin
    </a>
  </div>
</DocumentFragment>
`;
